src/goose/metrics.clj

```diff
@@ -33,6 +34,14 @@
   (increment [this key value tags] "Increment given key by value.")
   (timing [this key duration tags] "Record duration of given key."))
 
+;; nil behaves equivalent to a disabled metric.
+(extend-protocol Metrics
+  nil
+  (enabled? [_] false)
+  (gauge [_ _ _ _] (log/error "tried to gauge without a metrics plugin"))
+  (increment [_ _ _ _] (log/error "tried to increment without a metrics plugin"))
+  (timing [_ _ _ _] (log/error "tried to record duration without a metrics plugin")))
```

Comment:
I have looked through all the references to these methods being called, and can confirm that none of these are used without checking on `enabled?` first—not doing so is a bug anyway with other metrics plugin, so we log it as an error.